From 25d2386174bbf8c0f9e40f632f0e9c962e8d0284 Mon Sep 17 00:00:00 2001
From: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Date: Fri, 26 Apr 2024 15:30:55 +0200
Subject: [PATCH 1093/1695] dt-bindings: soc: add risal description

Add risal description in "stm32mp25-rif.h" and "stm32mp25-rifsc.h"

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 938b108ccf3cc28c96b49adf30d60d55971427a5 dt-bindings: soc: add risal description]
Change-Id: Iecd45590e9ebdd7dae50f238f8f1710f2a78b1c9
Signed-off-by: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/376882
Tested-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 .../include/dt-bindings/soc/stm32mp25-rifsc.h | 37 +++++++++++++++++++
 1 file changed, 37 insertions(+)

diff --git a/core/include/dt-bindings/soc/stm32mp25-rifsc.h b/core/include/dt-bindings/soc/stm32mp25-rifsc.h
index 08d589b5e..2055bd0b9 100644
--- a/core/include/dt-bindings/soc/stm32mp25-rifsc.h
+++ b/core/include/dt-bindings/soc/stm32mp25-rifsc.h
@@ -171,5 +171,42 @@
 	 ((mcid) << RIMUPROT_RIMC_MCID_SHIFT) | \
 	 ((mode) << RIMUPROT_RIMC_MODE_SHIFT))
 
+/* RISAL ID */
+#define RISAL_ID(idx)		(idx)
+
+/* RISAL CID subregion modes */
+#define RIFSC_RISAL_SRDIS		0x0
+#define RIFSC_RISAL_SREN		0x1
+
+/* RISAL Block id */
+#define RIFSC_RISAL_BLOCK_A		0x0
+#define RIFSC_RISAL_BLOCK_B		0x1
+
+#define RIFSC_RISAL_REG_ID_SHIFT	0
+#define RIFSC_RISAL_BLOCK_ID_SHIFT	8
+#define RIFSC_RISAL_REGx_CFGR_SHIFT	16
+#define RIFSC_RISAL_SREN_SHIFT	16
+#define RIFSC_RISAL_LOCK_SHIFT	17
+#define RIFSC_RISAL_SRCID_SHIFT	20
+#define RIFSC_RISAL_SEC_SHIFT		24
+#define RIFSC_RISAL_PRIV_SHIFT	25
+#define RIFSC_RISAL_REG_ID_MASK	GENMASK_32(7, 0)
+#define RIFSC_RISAL_BLOCK_ID_MASK	GENMASK_32(15, 8)
+#define RIFSC_RISAL_SRCID_MASK	GENMASK_32(22, 20)
+#define RIFSC_RISAL_REGx_CFGR_MASK	(BIT(RIFSC_RISAL_SREN_SHIFT) | \
+					 BIT(RIFSC_RISAL_LOCK_SHIFT) | \
+					 RIFSC_RISAL_SRCID_MASK | \
+					 BIT(RIFSC_RISAL_SEC_SHIFT) | \
+					 BIT(RIFSC_RISAL_PRIV_SHIFT))
+
+#define RISALPROT(risalid, blockid, srcid, lock, sec, priv, sren) \
+	 (((risalid) << RIFSC_RISAL_REG_ID_SHIFT) | \
+	  ((blockid) << RIFSC_RISAL_BLOCK_ID_SHIFT) | \
+	  ((priv) << RIFSC_RISAL_PRIV_SHIFT) | \
+	  ((sec) << RIFSC_RISAL_SEC_SHIFT) | \
+	  ((srcid) << RIFSC_RISAL_SRCID_SHIFT) | \
+	  ((lock) << RIFSC_RISAL_LOCK_SHIFT) | \
+	  ((sren) << RIFSC_RISAL_SREN_SHIFT))
+
 #endif /* _DT_BINDINGS_STM32_RIFSC_H */
 
-- 
2.47.2

